File: java/E2.java

public class E2 extends Thread {

    public E2(String n) {
        super(n);
    }

    public static void main(String[] args) {
        E2 t1 = new E2("1");E2 t2 = new E2("2");E2 t3 = new E2("3");E2 t4 = new E2("4");E2 t5 = new E2("5");
        t1.start();t2.start();t3.start();t4.start();t5.start();
    }

    @Override
    public void run() {
        System.out.println(Thread.currentThread().getName());
    }

}

File: python/E2.py
import threading
B = 500
lock = threading.Lock()
def updatebalance(c):
    lock.acquire()
    global B
    if (B + c < 1000):
        print(threading.currentThread().getName(),"Minimum balance")
    else:
        print(threading.currentThread().getName(),B)
        B = B + c
        print(threading.currentThread().getName(),B) 
    lock.release()
t = []
for i in range(1,11):
    if (i %2):
        t.append(threading.Thread(None,updatebalance,i,(-100,)))
    else:
        t.append(threading.Thread(None,updatebalance,i,(100,)))    
for i in t:
    i.start()